js-core/homeworks/phonebook/index.js

```diff
@@ -0,0 +1,106 @@
+class PhoneApp {
+	constructor() {
+		this.dataBase = [
+        	{id:1, name:'Vasya', phone:'qweqwe'},
+        	{id:6, name:'Vasya', phone:'qweqwe'},
+        	{id:3, name:'Vasya', phone:'qweqwe'},
+        	{id:8, name:'Vasya', phone:'qweqwe'},
+        	{id: 10, name: 'Tom2', phone: '0995385'},
+        	{id: 10, name: 'Anita', phone: '0995305385'},
+        	{id: 12, name: 'Anita', phone: '9995305385'}
+    	]
+	};
+
+	checkForNumbers(phone) {
+		let arrPhone = phone.split('');
+		let numberArrPhone = arrPhone.map((elem) => {
+			return parseInt(elem);
+		});
+		let validNumber = numberArrPhone.some((elem) => {
+			return isNaN(elem)
+		}); 
+		if(validNumber) {
+			// console.error('Телефонный номер должен содержать только цифры');
+			return false;
+		} else {
+			//console.log('Телефонный номер вылидный');
+			return true;
+		};
+	};
+
+	transformNumber(phone) {
+		if(this.checkForNumbers(phone)){
+			let arrPhone = phone.split('');
+			arrPhone.splice(0, 0, '(');
+			arrPhone.splice(4, 0, ')', ' ');
+			arrPhone.splice(8, 0, '-');
+			arrPhone.splice(11, 0, '-');
+			return arrPhone.join('');
+		} else {
+			return 'Телефонный номер должен содержать только цифры';
+		};
+	};
+
+	addUser(id, name, phone) {
+		let objUser = {
```

Comment:
You can create an additional class which would create such a user.

And if we go further maybe even phone-number validation should be made inside of User class.

Not sure if it the responsibility of phone-book to validate users.
The main responsibility of phone-book is to find users, update them and delete them